src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java

```diff
@@ -0,0 +1,181 @@
+
+        TalonFXConfiguration LEFT_TALONFX_CONFIG = new TalonFXConfiguration();
+        TalonFXConfiguration RIGHT_TALONFX_CONFIG = new TalonFXConfiguration();
+
+        // TODO: Check On Robot
+        
+
+        LEFT_TALONFX_CONFIG.Slot0.kS = kS;
+        LEFT_TALONFX_CONFIG.Slot0.kV = kV;
+        LEFT_TALONFX_CONFIG.Slot0.kA = kA;
+
+        LEFT_TALONFX_CONFIG.Slot0.kP = kP;
+        LEFT_TALONFX_CONFIG.Slot0.kI = kI;
+        LEFT_TALONFX_CONFIG.Slot0.kD = kD;
+
+        RIGHT_TALONFX_CONFIG = LEFT_TALONFX_CONFIG;
+
+        
+        //TODO: CHECK DIRECTIONS
+        LEFT_TALONFX_CONFIG.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        RIGHT_TALONFX_CONFIG.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+
+        leftMotor.getConfigurator().apply(LEFT_TALONFX_CONFIG);
+        rightMotor.getConfigurator().apply(RIGHT_TALONFX_CONFIG);
+
+        // will remove when shooter is tested
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * sets the target velocity for the left motor in rotations per second
+     * 
+     * @param velocity in rotations per second
+     */
+    public void setLeftVelocity(double velocity) {
+        leftMotionMagicVelocityVoltage.Velocity = velocity;
+        leftMotor.setControl(leftMotionMagicVelocityVoltage);
+    }
+
+    /**
+     * sets the target velocity for the right motor in rotations per second
+     * 
+     * @param velocity in rotations per second
+     */
+    public void setRightVelocity(double velocity) {
+        rightMotionMagicVelocityVoltage.Velocity = velocity;
+        rightMotor.setControl(rightMotionMagicVelocityVoltage);
+    }
+
+    /**
+     * returns the difference between the left motor's current velocity and the
+     * target velocity
+     * 
+     * @return velocity error in rotations per second
+     */
+    private double getLeftVelocityError() {
+        return leftMotor.getVelocity().getValueAsDouble() - leftMotionMagicVelocityVoltage.Velocity;
+    }
+
+    /**
+     * returns the difference between the right motor's current velocity and the
+     * target velocity
+     * 
+     * @return velocity error in rotations per second
+     */
+    private double getRightVelocityError() {
+        return rightMotor.getVelocity().getValueAsDouble() - rightMotionMagicVelocityVoltage.Velocity;
+    }
+
+    public void stopShooterMotors() {
+        rightMotor.setControl(new CoastOut());
+        leftMotor.setControl(new CoastOut());
+    }
+
+    /**
+     * returns if both motors are within the velocity tolerance
+     * 
+     * @return true if both motors are within the velocity tolerance
+     */
+    public boolean motorsAtTargetVelocity() {
+        return Math.abs(getLeftVelocityError()) <= VEOLOCITY_TOLERANCE
+                && Math.abs(getRightVelocityError()) <= VEOLOCITY_TOLERANCE;
+    }
+
+    // getters and setters for pid constants thru shuffleboard
+    public double getkS() {
```

Comment:
I bet you don't want these to be public :)

Also, all these little getters and setters are annoying, aren't they? Could you have used lambda's below to have them inline?

Reply:
```builder.addDoubleProperty("kS", () -> this.kS, (v) -> this.kS = v);```

Reply:
Yup, was just about to comment about how you can have lambdas with parameters, since for the get functions you are interfacing with something called a `DoubleConsumer`